connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -0,0 +1,195 @@
+    private static final String REST_HOST_NAME = "localhost";
+
+    private final EmbeddedKafkaCluster kafkaCluster;
+
+    private final Map<String, String> workerProps;
+    private final String clusterName;
+
+    private Connect connect;
+
+    public EmbeddedConnectCluster(Class<?> klass) {
+        this(klass.getName());
+    }
+
+    public EmbeddedConnectCluster(String name) {
+        // this empty map will be populated with defaults before starting Connect.
+        this(name, new HashMap<>());
+    }
+
+    public EmbeddedConnectCluster(String name, Map<String, String> workerProps) {
+        this(name, workerProps, DEFAULT_NUM_BROKERS, DEFAULT_BROKER_CONFIG);
+    }
+
+    public EmbeddedConnectCluster(String name, Map<String, String> workerProps, int numBrokers, Properties brokerProps) {
+        this.workerProps = workerProps;
+        this.clusterName = name;
+        kafkaCluster = new EmbeddedKafkaCluster(numBrokers, brokerProps);
+    }
+
+    public void start() throws IOException {
+        before();
+    }
+
+    /**
+     * Stop the connect cluster and the embedded Kafka and
+     */
+    public void stop() {
+        after();
+    }
+
+    @Override
+    protected void before() throws IOException {
+        kafkaCluster.before();
+        startConnect();
+    }
+
+    @Override
+    protected void after() {
+        try {
+            connect.stop();
+        } catch (Exception e) {
+            log.error("Could not stop connect", e);
+        }
+
+        try {
+            kafkaCluster.after();
+        } catch (Exception e) {
+            log.error("Could not stop kafka", e);
+        }
+    }
+
+    public void startConnect() {
+        log.info("Starting Connect cluster with one worker.");
+
+        workerProps.put(BOOTSTRAP_SERVERS_CONFIG, kafka().bootstrapServers());
+        workerProps.put(REST_HOST_NAME_CONFIG, REST_HOST_NAME);
+
+        putIfAbsent(workerProps, GROUP_ID_CONFIG, "connect-integration-test-" + clusterName);
+        putIfAbsent(workerProps, OFFSET_STORAGE_TOPIC_CONFIG, "connect-offset-topic-" + clusterName);
+        putIfAbsent(workerProps, OFFSET_STORAGE_REPLICATION_FACTOR_CONFIG, "1");
+        putIfAbsent(workerProps, CONFIG_TOPIC_CONFIG, "connect-config-topic-" + clusterName);
+        putIfAbsent(workerProps, CONFIG_STORAGE_REPLICATION_FACTOR_CONFIG, "1");
+        putIfAbsent(workerProps, STATUS_STORAGE_TOPIC_CONFIG, "connect-storage-topic-" + clusterName);
+        putIfAbsent(workerProps, STATUS_STORAGE_REPLICATION_FACTOR_CONFIG, "1");
+        putIfAbsent(workerProps, KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
+        putIfAbsent(workerProps, VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.storage.StringConverter");
+
+        connect = new ConnectDistributed().startConnect(workerProps);
+    }
+
+    public void startConnector(String connName, Map<String, String> connConfig) throws IOException {
```

Comment:
Might be nice to have an `updateConnector` method, since I'd probably want to have tests that check the behavior (at least within AK) to verify that connectors can be updated and they (re)start correctly.

Reply:
Yeah, what if I want to test updating a connector configuration to see if my connector responds correctly? Seems valuable enough, and if it's not in the integration test framework connector projects simply can't do it. (They could implement their own, but why not have it here?)